### PR TITLE
Allows std.traits.getUDAs to filter attributes by template type.

### PR DESCRIPTION
**Pull request by japplegame**
See unittests.

### CODE REVIEW

File: std/traits.d [hunk 1]
```diff
@@ -6698,6 +6704,22 @@ unittest
     static assert(getUDAs!(c, Attr)[0].value == 42);
     static assert(getUDAs!(c, Attr)[1].name == "Pi");
     static assert(getUDAs!(c, Attr)[1].value == 3);
+
+    struct AttrT(T)
+    {
+        string name;
+        int value;
```
Comment (Hackerpilot) [inline std/traits.d]: Shouldn't this be `T`?

Comment (japplegame) [inline std/traits.d]: Does it really matter?

Comment (Hackerpilot) [inline std/traits.d]: For the correctness of the unit test? No. For clarity? Yes. I'd prefer not having to explain that the "T" in this struct declaration has no purpose.

Comment (japplegame) [inline std/traits.d]: OK

**Comment by Hackerpilot** [comment]
Auto-merge toggled on

**Pull request merged**